core/http/src/header/accept.rs

```diff
@@ -146,24 +146,22 @@ impl Accept {
         let mut all = self.iter();
         let mut preferred = all.next().unwrap_or(&ANY);
         for media_type in all {
-            if media_type.weight().is_none() && preferred.weight().is_some() {
```

Comment:
Not only is this less legible than the previous, but it is also different logically as the previous version short-circuited and this one does not. Let's revert this.

Reply:
The logic did not change because the content of the if is the same (see [if_same_then_else](https://rust-lang.github.io/rust-clippy/master/index.html#if_same_then_else)). Here's the allow I've added :
```
    #[allow(clippy::if_same_then_else)] // For this particular case, we need to keep the if/else
                                        // branching for legibility.
```
If it is not normal to have the same content inside theses if, we will need to create an issue to fix this function properly.

Reply:
By "logically" I mean that they compute differently. The version you've written computes every component of the `||` on every loop whereas the previous version only computes up to the first that matches. This is fundamentally different.

There is no such thing as "normal". The code was written the original way for a reason: to more clearly explain what it is doing. There is nothing to fix. The version you've written makes it much more difficult, in my opinion, to see this, and has the side effect of being more expensive to compute in expectation.

Reply:
Thank you Sergio for the explanation. I'll update the comment to better explain the initial intent:
```
    #[allow(clippy::if_same_then_else)] // For this particular case, we need to keep the if/else
                                        // branching for legibility of comments and to avoid
                                        // unnecessary computation.
```